public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx

```diff
@@ -0,0 +1,410 @@
+import { stateToColorMap } from '../../../utils/constants';
+import {
+  HISTORICAL_DETECTOR_RESULT_REFRESH_RATE,
+  HISTORICAL_DETECTOR_ACTION,
+  HISTORICAL_DETECTOR_STOP_THRESHOLD,
+} from '../../utils/constants';
+import {
+  getCallout,
+  waitForMs,
+  getRunningHistoricalDetectorCount,
+} from '../../utils/helpers';
+import { CoreStart } from '../../../../../../../src/core/public';
+import { CoreServicesContext } from '../../../../components/CoreServices/CoreServices';
+import { getClusterStats } from '../../../../redux/reducers/elasticsearch';
+import { GET_ALL_DETECTORS_QUERY_PARAMS } from '../../../utils/constants';
+
+export interface HistoricalDetectorRouterProps {
+  detectorId?: string;
+}
+interface HistoricalDetectorDetailProps
+  extends RouteComponentProps<HistoricalDetectorRouterProps> {}
+
+interface HistoricalDetectorDetailModalState {
+  isOpen: boolean;
+  action: HISTORICAL_DETECTOR_ACTION | undefined;
+}
+
+export const HistoricalDetectorDetail = (
+  props: HistoricalDetectorDetailProps
+) => {
+  const core = React.useContext(CoreServicesContext) as CoreStart;
+  const isDark = darkModeEnabled();
+  useHideSideNavBar(true, false);
+  const dispatch = useDispatch();
+  const detectorId: string = get(props, 'match.params.detectorId', '');
+
+  const adState = useSelector((state: AppState) => state.ad);
+  const allDetectors = adState.detectors;
+  const errorGettingDetectors = adState.errorMessage;
+  const detector = allDetectors[detectorId];
+  const monitors = useSelector((state: AppState) => state.alerting.monitors);
+  const monitor = get(monitors, `${detectorId}.0`);
+
+  // TODO: look further into whether or not we need to check for missing data or not
+  const isDetectorMissingData = false;
+
+  const [isLoadingDetector, setIsLoadingDetector] = useState<boolean>(true);
+  const [isStoppingDetector, setIsStoppingDetector] = useState<boolean>(false);
+  const [
+    historicalDetectorDetailModalState,
+    setHistoricalDetectorDetailModalState,
+  ] = useState<HistoricalDetectorDetailModalState>({
+    isOpen: false,
+    action: undefined,
+  });
+  const callout = getCallout(detector, isStoppingDetector);
+
+  useEffect(() => {
+    if (errorGettingDetectors) {
+      core.notifications.toasts.addDanger(
+        'Unable to get the historical detector'
+      );
+      props.history.push('/historical-detectors');
+    }
+  }, [errorGettingDetectors]);
+
+  useEffect(() => {
+    if (detector) {
+      core.chrome.setBreadcrumbs([
+        BREADCRUMBS.ANOMALY_DETECTOR,
+        BREADCRUMBS.HISTORICAL_DETECTORS,
+        { text: detector ? detector.name : '' },
+      ]);
+    }
+  }, [detector]);
+
+  const fetchDetector = async () => {
+    try {
+      await dispatch(getDetector(detectorId));
+      setIsLoadingDetector(false);
+    } catch {
+      core.notifications.toasts.addDanger(
+        'Error retrieving the historical detector'
+      );
```

Comment:
If can't retrieve historical detector, should we call `setIsLoadingDetector(true)`  and redirect to historical detectors list?

Reply:
After looking at this, I realized we don't need any toast here at all, since it will be caught in the custom hook on line 102 [here](https://github.com/ohltyler/anomaly-detection-kibana-plugin/blob/add-details-page/public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx#L102)., where a toast is displayed and the user is redirected to the historical detectors list. Will remove this catch block altogether.

Reply:
And seems we would want to call `setIsLoadingDetector(false)` here instead?

Reply:
After looking further, the entire `isLoadingDetector` field isn't used at all. Will remove this line entirely.